File: system/src/main/java/com/ticketing/system/service/AdminService.java
package com.ticketing.system.service;

import java.util.List;
import java.util.stream.Collectors;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.ticketing.system.dto.RegisterRequest;
import com.ticketing.system.dto.UserResponse;
import com.ticketing.system.entity.Role;
import com.ticketing.system.entity.User;
import com.ticketing.system.exception.ResourceNotFoundException;
import com.ticketing.system.repository.UserRepository;

@Service
public class AdminService {
    
    @Autowired
    private UserRepository userRepository;
    
    @Autowired
    private PasswordEncoder passwordEncoder;
    
    public List<UserResponse> getAllUsers() {
        return userRepository.findAll().stream()
                .map(this::convertToUserResponse)
                .collect(Collectors.toList());
    }
    
    @Transactional
    public UserResponse createUser(RegisterRequest request) {
        if (userRepository.existsByEmail(request.getEmail())) {
            throw new RuntimeException("Email already exists");
        }
        
        User user = new User();
        user.setEmail(request.getEmail());
        user.setPassword(passwordEncoder.encode(request.getPassword()));
        user.setName(request.getName());
        user.setRole(request.getRole() != null ? request.getRole() : Role.USER);
        
        user = userRepository.save(user);
        return convertToUserResponse(user);
    }
    
    @Transactional
    public UserResponse updateUserRole(Long userId, Role role) {
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new ResourceNotFoundException("User not found"));
        
        user.setRole(role);
        user = userRepository.save(user);
        
        return convertToUserResponse(user);
    }
    
    @Transactional
    public void deleteUser(Long userId) {
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new ResourceNotFoundException("User not found"));
        
        userRepository.delete(user);
    }
    
    public List<UserResponse> getUsersByRole(Role role) {
        return userRepository.findByRole(role).stream()
                .map(this::convertToUserResponse)
                .collect(Collectors.toList());
    }
    
    private UserResponse convertToUserResponse(User user) {
        UserResponse response = new UserResponse();
        response.setId(user.getId());
        response.setEmail(user.getEmail());
        response.setName(user.getName());
        response.setRole(user.getRole());
        return response;
    }
}

File: frontend/ticketing-frontend/src/services/adminService.ts
import axiosInstance from '@/lib/axios';

export interface CreateUserData {
  email: string;
  password: string;
  name: string;
  role: 'USER' | 'SUPPORT_AGENT' | 'ADMIN';
}

export const adminService = {
  getAllUsers: async () => {
    const response = await axiosInstance.get('/admin/users');
    return response.data;
  },

  createUser: async (data: CreateUserData) => {
    const response = await axiosInstance.post('/admin/users', data);
    return response.data;
  },

  updateUserRole: async (id: number, role: string) => {
    const response = await axiosInstance.put(`/admin/users/${id}/role?role=${role}`);
    return response.data;
  },

  deleteUser: async (id: number) => {
    const response = await axiosInstance.delete(`/admin/users/${id}`);
    return response.data;
  },

  getUsersByRole: async (role: string) => {
    const response = await axiosInstance.get(`/admin/users/role/${role}`);
    return response.data;
  },
};